certbot/tests/cli_test.py

```diff
@@ -479,6 +479,16 @@ def test_route53_no_revert(self):
             for topic in ['all', 'plugins', 'dns-route53']:
                 self.assertNotIn('certbot-route53:auth', self._help_output([help_flag, topic]))
 
+    def test_config_option(self):
+        namespace = self.parse([])
+        self.assertIsNone(namespace.config_file)
+        with tempfile.NamedTemporaryFile() as tmp_config:
+            tmp_config.close()  # close now because of compatibility issues on Windows
+            namespace = self.parse(['-c', tmp_config.name])
+            self.assertEqual(namespace.config_file, tmp_config.name)
+            namespace = self.parse(['--config', tmp_config.name])
+            self.assertEqual(namespace.config_file, tmp_config.name)
```

Comment:
With this test failure, `tempfile.NamedTemporaryFile()` deletes the file as soon as `close` is called.

I think that other test `test_cli_ini_domains`, is probably buggy. The file doesn't exist until the `open` call and it never cleans it up either.

Maybe you could disable the auto-deletion, and instead remove the file in a `finally` clause?

```suggestion
        tmp_config = tempfile.NamedTemporaryFile(delete=False)
        try:
            tmp_config.close()  # close now because of compatibility issues on Windows
            namespace = self.parse(['-c', tmp_config.name])
            self.assertEqual(namespace.config_file, tmp_config.name)
            namespace = self.parse(['--config', tmp_config.name])
            self.assertEqual(namespace.config_file, tmp_config.name)
        finally:
            os.remove(tmp_config.name)
```

Reply:
This fixes the Linux errors of my latest attempt (thanks!), but I'm wondering if this will fix the Windows permission error as seen in caaff69. Only one way to find out, as I don't have Windows running to check that.

Reply:
Seems that did the job indeed! Nice!

Although my [previous point](https://github.com/certbot/certbot/pull/8806#discussion_r646034577) still applies: this test doesn't really check if the default configuration files are actually *gone*, as those info isn't available in `namespace`..